source/solvers/fluid_dynamics_matrix_free.cc

```diff
@@ -2440,14 +2439,11 @@ FluidDynamicsMatrixFree<dim>::set_initial_condition_fd(
         solve_non_linear_system(false);
       this->finish_time_step();
 
-      // Set the kinematic viscosity in the system operator to be the original
-      // viscosity
-      viscosity_model->set_kinematic_viscosity(viscosity_end);
-
-      // Reset kinematic viscosity to simulation parameters
-      viscosity_model->set_kinematic_viscosity(viscosity_end);
-      this->system_operator->set_kinematic_viscosity(viscosity_end);
+      // Reset original rheology for the system operator
+      this->system_operator->properties_manager->set_rheology(
+        original_viscosity_model);
```

Comment:
Why not set just the property manager of the fluid dynamics class, I'm not sure i understand why you are accessing the one in the operator to do that?